auth_oidc/models/auth_oauth_provider.py

```diff
@@ -50,6 +50,46 @@ class AuthOauthProvider(models.Model):
         help="Additional parameters for the auth link. "
         "For example: {'prompt':'select_account'}"
     )
+    oidc_discovery_url = fields.Char(string="OIDC config URL")
+    supports_oidc_discovery = fields.Boolean(default=False)
+    auth_endpoint = fields.Char(required=False)
+    scopes_supported = fields.Char()
+
+    @api.constrains("scope")
+    def constrains_available_scopes(self):
+        if not self.scopes_supported:
+            return
+        for scope in self.scope.split(" "):
+            if ":" in scope:
+                while len(scope) > 1:
+                    scope = ":".join(scope.split(":")[0:-1])
+                    if scope + ":*" not in self.scopes_supported:
+                        raise exceptions.ValidationError(
+                            _("Scope %(scope)s is not supported by provider")
+                            % {"scope": scope}
+                        )
+            if scope not in self.scopes_supported:
+                raise exceptions.ValidationError(
+                    _("Scope %(scope)s is not supported by provider") % {"scope": scope}
+                )
+
+    def action_get_oidc_configuration(self):
+        if not self.supports_oidc_discovery:
+            return
+        r = requests.get(self.oidc_discovery_url, timeout=10)
+        r.raise_for_status()
+        response = r.json()
+        try:
+            self.auth_endpoint = response["authorization_endpoint"]
+            self.validation_endpoint = response["userinfo_endpoint"]
+        except KeyError:
+            raise exceptions.ValidationError(
+                _("Incorrect response from issuer, check OIDC config URL")
+            ) from KeyError
+        self.jwks_uri = response.get("jwks_uri", False)
+        self.scopes_supported = response.get("scopes_supported", False)
```

Comment:
According to https://openid.net/specs/openid-connect-discovery-1_0.html, scopes_supported is a json array, so this should be converted to a string?